File: frontend/apps/web-antd/src/api/workbench/scheduled-report.ts
/**
 * 定时报表 API
 */
import { requestClient } from '#/api/request';

import type { PageResult } from '../matter/types';

// ========== 类型定义 ==========
export interface ScheduledReportDTO {
  id: number;
  taskNo?: string;
  taskName: string;
  description?: string;
  templateId: number;
  templateName?: string;
  scheduleType: string;
  scheduleTypeName?: string;
  cronExpression?: string;
  executeTime?: string;
  executeDayOfWeek?: number;
  executeDayOfMonth?: number;
  scheduleDescription?: string;
  reportParameters?: Record<string, any>;
  outputFormat?: string;
  notifyEnabled?: boolean;
  notifyEmails?: string[];
  notifyUserIds?: number[];
  status?: string;
  statusName?: string;
  lastExecuteTime?: string;
  lastExecuteStatus?: string;
  lastExecuteStatusName?: string;
  nextExecuteTime?: string;
  totalExecuteCount?: number;
  successCount?: number;
  failCount?: number;
  createdBy?: number;
  createdByName?: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface ScheduledReportLogDTO {
  id: number;
  taskId: number;
  taskNo?: string;
  executeTime?: string;
  status?: string;
  statusName?: string;
  reportId?: number;
  fileUrl?: string;
  fileSize?: number;
  fileSizeDisplay?: string;
  durationMs?: number;
  durationDisplay?: string;
  errorMessage?: string;
  notifyStatus?: string;
  notifyStatusName?: string;
  notifyResult?: string;
  createdAt?: string;
}

export interface ScheduledReportQuery {
  pageNum?: number;
  pageSize?: number;
  keyword?: string;
  status?: string;
}

export interface CreateScheduledReportCommand {
  taskName: string;
  description?: string;
  templateId: number;
  scheduleType: string;
  cronExpression?: string;
  executeTime?: string;
  executeDayOfWeek?: number;
  executeDayOfMonth?: number;
  reportParameters?: Record<string, any>;
  outputFormat?: string;
  notifyEnabled?: boolean;
  notifyEmails?: string[];
  notifyUserIds?: number[];
}

export interface ScheduledReportLogQuery {
  pageNum?: number;
  pageSize?: number;
  status?: string;
}

// ========== API 函数 ==========

/** 分页查询定时报表任务 */
export function getScheduledReportList(params: ScheduledReportQuery) {
  return requestClient.get<PageResult<ScheduledReportDTO>>('/workbench/scheduled-report', { params });
}

/** 获取定时任务详情 */
export function getScheduledReportDetail(id: number) {
  return requestClient.get<ScheduledReportDTO>(`/workbench/scheduled-report/${id}`);
}

/** 创建定时报表任务 */
export function createScheduledReport(data: CreateScheduledReportCommand) {
  return requestClient.post<ScheduledReportDTO>('/workbench/scheduled-report', data);
}

/** 更新定时报表任务 */
export function updateScheduledReport(id: number, data: CreateScheduledReportCommand) {
  return requestClient.put<ScheduledReportDTO>(`/workbench/scheduled-report/${id}`, data);
}

/** 删除定时报表任务 */
export function deleteScheduledReport(id: number) {
  return requestClient.delete(`/workbench/scheduled-report/${id}`);
}

/** 启用任务 */
export function enableScheduledReport(id: number) {
  return requestClient.post(`/workbench/scheduled-report/${id}/enable`);
}

/** 暂停任务 */
export function pauseScheduledReport(id: number) {
  return requestClient.post(`/workbench/scheduled-report/${id}/pause`);
}

/** 立即执行任务 */
export function executeScheduledReportNow(id: number) {
  return requestClient.post<any>(`/workbench/scheduled-report/${id}/execute`);
}

/** 查询执行记录 */
export function getScheduledReportLogs(id: number, params: ScheduledReportLogQuery) {
  return requestClient.get<PageResult<ScheduledReportLogDTO>>(`/workbench/scheduled-report/${id}/logs`, { params });
}

File: backend/src/main/java/com/lawfirm/domain/workbench/entity/ScheduledReport.java
package com.lawfirm.domain.workbench.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.lawfirm.common.base.BaseEntity;
import java.time.LocalDateTime;
import lombok.*;
import lombok.experimental.SuperBuilder;

/** 定时报表任务实体 */
@Data
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@TableName("workbench_scheduled_report")
public class ScheduledReport extends BaseEntity {

  /** 任务编号 */
  private String taskNo;

  /** 任务名称 */
  private String taskName;

  /** 任务描述 */
  private String description;

  /** 关联的报表模板ID */
  private Long templateId;

  /** 调度类型：DAILY-每日, WEEKLY-每周, MONTHLY-每月, CRON-自定义 */
  private String scheduleType;

  /** Cron表达式 */
  private String cronExpression;

  /** 执行时间 HH:mm */
  private String executeTime;

  /** 执行星期几 1-7 */
  private Integer executeDayOfWeek;

  /** 执行日期 1-31 */
  private Integer executeDayOfMonth;

  /** 报表参数（JSON格式） */
  private String reportParameters;

  /** 输出格式：EXCEL, PDF */
  private String outputFormat;

  /** 是否启用通知 */
  private Boolean notifyEnabled;

  /** 通知邮箱列表 */
  private String notifyEmails;

  /** 通知用户ID列表 */
  private String notifyUserIds;

  /** 状态：ACTIVE-启用, PAUSED-暂停, INACTIVE-停用 */
  private String status;

  /** 上次执行时间 */
  private LocalDateTime lastExecuteTime;

  /** 上次执行状态 */
  private String lastExecuteStatus;

  /** 下次执行时间 */
  private LocalDateTime nextExecuteTime;

  /** 总执行次数 */
  private Integer totalExecuteCount;

  /** 成功次数 */
  private Integer successCount;

  /** 失败次数 */
  private Integer failCount;

  /** 创建人姓名 */
  private String createdByName;
}
